WooCommerce/Classes/Model/ProductVariation+Woo.swift

```diff
@@ -0,0 +1,31 @@
+import Foundation
+import Yosemite
+
+/// Yosemite.ProductVariation helper methods
+///
+extension ProductVariation {
+
+    /// Generates a name for the product variation, given a list of the parent product attributes, e.g. "Blue - Any Size"
+    /// - Parameter allAttributes: A list of attributes from the parent `Product`
+    ///
+    func generateVariationName(from allAttributes: [ProductAttribute]) -> String {
+        let variationAttributes = generateVariationAttributes(from: allAttributes)
+        return variationAttributes.map { $0.nameOrValue }.joined(separator: " - ")
+    }
+
+    /// Generates the variation attributes, given a list of the parent product attributes.
+    /// - Parameter allAttributes: A list of attributes from the parent `Product`
+    ///
+    func generateVariationAttributes(from allAttributes: [ProductAttribute]) -> [VariationAttributeViewModel] {
+        return allAttributes
+            .sorted(by: { (lhs, rhs) -> Bool in
+                lhs.position < rhs.position
+            })
+            .map { productAttribute -> VariationAttributeViewModel in
+            guard let variationAttribute = attributes.first(where: { $0.id == productAttribute.attributeID && $0.name == productAttribute.name }) else {
+                return VariationAttributeViewModel(name: productAttribute.name)
+            }
+            return VariationAttributeViewModel(productVariationAttribute: variationAttribute)
+        }
+    }
+}
```

Comment:
A small nit here is that this logic does not seem suitable for a model extension, it seems to me that it is pure presentation logic, which should be part of the view model domain. 🤔 

Maybe a dedicated type named along the lines of `ProductVariationFormatter` makes more sense, what do you think?

Reply:
Yeah, that makes sense. I changed this in [add1b67](https://github.com/woocommerce/woocommerce-ios/pull/6131/commits/add1b6767db1a245ee0375dad99debe639716384); it changes quite a few lines but essentially just moves everything here from the extension to a `ProductVariationFormatter` type (and adds tests for that type).